2024-07-11 12:28:19.735896: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:28:19.736093: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:28:28,263 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:28:28,369 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:28:28,369 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:28:28,369 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:28:29,568 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:28:29,568 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:29,585 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.46}
INFO flwr 2024-07-11 12:28:29,586 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:28:29,764 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:29,785 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:28:29,788 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:29,796 | server.py:131 | fit progress: (1, 0.39399058053360125, {'accuracy': 0.8233333333333334}, 0.2105511)
DEBUG flwr 2024-07-11 12:28:29,796 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:29,818 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:28:29,819 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:28:29,819 | server.py:227 | fit_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,843 | server.py:241 | fit_round 2 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:29,854 | server.py:131 | fit progress: (2, 0.39186229534874445, {'accuracy': 0.84}, 0.2682524999999991)
DEBUG flwr 2024-07-11 12:28:29,854 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,876 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:29,876 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,900 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:29,911 | server.py:131 | fit progress: (3, 0.3917828389168051, {'accuracy': 0.84}, 0.32493260000000035)
DEBUG flwr 2024-07-11 12:28:29,911 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,932 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:29,932 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,961 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:29,971 | server.py:131 | fit progress: (4, 0.39184922231747926, {'accuracy': 0.84}, 0.3850436999999989)
DEBUG flwr 2024-07-11 12:28:29,972 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:29,993 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:29,993 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:30,020 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,030 | server.py:131 | fit progress: (5, 0.3917814757369386, {'accuracy': 0.84}, 0.44411949999999933)
DEBUG flwr 2024-07-11 12:28:30,030 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:30,051 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:30,051 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:30,079 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,090 | server.py:131 | fit progress: (6, 0.39185241193229176, {'accuracy': 0.84}, 0.5043644)
DEBUG flwr 2024-07-11 12:28:30,090 | server.py:178 | evaluate_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:30,111 | server.py:192 | evaluate_round 6 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:30,111 | server.py:227 | fit_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:30,142 | server.py:241 | fit_round 7 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,154 | server.py:131 | fit progress: (7, 0.3917816708283754, {'accuracy': 0.84}, 0.5678667999999991)
DEBUG flwr 2024-07-11 12:28:30,154 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:30,178 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:28:30,178 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:30,214 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,227 | server.py:131 | fit progress: (8, 0.391847444578247, {'accuracy': 0.8366666666666667}, 0.6406805999999996)
DEBUG flwr 2024-07-11 12:28:30,227 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:30,251 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:28:30,251 | server.py:227 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:30,287 | server.py:241 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,300 | server.py:131 | fit progress: (9, 0.3918664935268512, {'accuracy': 0.8366666666666667}, 0.714240199999999)
DEBUG flwr 2024-07-11 12:28:30,300 | server.py:178 | evaluate_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:30,326 | server.py:192 | evaluate_round 9 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:28:30,327 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:30,372 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:30,385 | server.py:131 | fit progress: (10, 0.39153574581884165, {'accuracy': 0.84}, 0.7985691999999993)
DEBUG flwr 2024-07-11 12:28:30,385 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:30,413 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:28:30,413 | server.py:153 | FL finished in 0.8269526000000003
INFO flwr 2024-07-11 12:28:30,414 | app.py:225 | app_fit: losses_distributed [(1, 0.3959679454565048), (2, 0.39536771873633064), (3, 0.3953618913888931), (4, 0.3953673134247462), (5, 0.39536179701487223), (6, 0.39536739389101666), (7, 0.3960705440619896), (8, 0.3956377624437727), (9, 0.39563769307629815), (10, 0.3948306939431599)]
INFO flwr 2024-07-11 12:28:30,414 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:28:30,414 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:28:30,414 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.39399058053360125), (2, 0.39186229534874445), (3, 0.3917828389168051), (4, 0.39184922231747926), (5, 0.3917814757369386), (6, 0.39185241193229176), (7, 0.3917816708283754), (8, 0.391847444578247), (9, 0.3918664935268512), (10, 0.39153574581884165)]
INFO flwr 2024-07-11 12:28:30,414 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.46), (1, 0.8233333333333334), (2, 0.84), (3, 0.84), (4, 0.84), (5, 0.84), (6, 0.84), (7, 0.84), (8, 0.8366666666666667), (9, 0.8366666666666667), (10, 0.84)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.09195402,  1.12497359,  0.82444381,  0.56609013,  0.2895109 ,
         0.10761589, -0.03355835,  0.25300674]]), array([-0.53828672])]
[array([[-2.09195402,  1.12497359,  0.82444381,  0.56609013,  0.2895109 ,
         0.10761589, -0.03355835,  0.25300674]]), array([-0.53828672])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01474524,  1.0091148 ,  0.79555331,  0.46720365,  0.26943477,
         0.13163001, -0.05882226,  0.04138257]]), array([-0.0866625])]
[array([[-2.01474524,  1.0091148 ,  0.79555331,  0.46720365,  0.26943477,
         0.13163001, -0.05882226,  0.04138257]]), array([-0.0866625])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01865079,  1.01143844,  0.79632787,  0.46777855,  0.26941646,
         0.13159467, -0.05967647,  0.04245788]]), array([-0.08732388])]
[array([[-2.01865079,  1.01143844,  0.79632787,  0.46777855,  0.26941646,
         0.13159467, -0.05967647,  0.04245788]]), array([-0.08732388])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01487081,  1.00987518,  0.79575271,  0.4672064 ,  0.26961309,
         0.13148459, -0.05897354,  0.04149886]]), array([-0.08772206])]
[array([[-2.01487081,  1.00987518,  0.79575271,  0.4672064 ,  0.26961309,
         0.13148459, -0.05897354,  0.04149886]]), array([-0.08772206])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01872302,  1.01146674,  0.79634533,  0.4677926 ,  0.26940929,
         0.1315866 , -0.05966235,  0.04244699]]), array([-0.08730328])]
[array([[-2.01872302,  1.01146674,  0.79634533,  0.4677926 ,  0.26940929,
         0.1315866 , -0.05966235,  0.04244699]]), array([-0.08730328])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01480081,  1.00979128,  0.79573472,  0.46719845,  0.26963327,
         0.13156011, -0.05897384,  0.04149459]]), array([-0.08776196])]
[array([[-2.01480081,  1.00979128,  0.79573472,  0.46719845,  0.26963327,
         0.13156011, -0.05897384,  0.04149459]]), array([-0.08776196])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01871301,  1.01146275,  0.79634254,  0.46778958,  0.26940889,
         0.13158612, -0.05966407,  0.04244936]]), array([-0.08730196])]
[array([[-2.01871301,  1.01146275,  0.79634254,  0.46778958,  0.26940889,
         0.13158612, -0.05966407,  0.04244936]]), array([-0.08730196])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03530177,  1.03438037,  0.82348709,  0.44272104,  0.2226348 ,
         0.11071473, -0.07453173,  0.0110257 ]]), array([0.0341065])]
[array([[-2.03530177,  1.03438037,  0.82348709,  0.44272104,  0.2226348 ,
         0.11071473, -0.07453173,  0.0110257 ]]), array([0.0341065])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03517174,  1.03435554,  0.82375023,  0.44277741,  0.22248742,
         0.11101395, -0.0744112 ,  0.0112293 ]]), array([0.03355485])]
[array([[-2.03517174,  1.03435554,  0.82375023,  0.44277741,  0.22248742,
         0.11101395, -0.0744112 ,  0.0112293 ]]), array([0.03355485])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05368325,  1.0641535 ,  0.82688732,  0.43369629,  0.21281731,
         0.10229428, -0.05979829,  0.02230719]]), array([0.00688366])]
[array([[-2.05368325,  1.0641535 ,  0.82688732,  0.43369629,  0.21281731,
         0.10229428, -0.05979829,  0.02230719]]), array([0.00688366])]
>>>>>>>>>>>>>run time:
0.5s
